src/wechaty/user/friendship.py

```diff
@@ -138,33 +134,31 @@ def __str__(self) -> str:
             # TODO -> get constructor name of the friendship
             return 'Friendship'
         return 'Friendship # {0} <{1}>'.format(
-            str(self._payload.type),
-            self.payload.contact_id)
+            str(self._payload.type), self._payload.contact_id)
 
     def is_ready(self) -> bool:
         """
         check if friendship is ready
         """
-        return self.puppet is None or self.payload is None
+        return self.puppet is not None and self.payload is not None
 
     async def ready(self):
         """
         load friendship payload
         """
-        if not self.is_ready():
-            friendship_search_response = await self.puppet.friendship_payload(
-                friendship_id=self.friendship_id)
-            self._payload = friendship_search_response
-        if self.payload is None:
-            raise Exception('can"t not load friendship payload %s'
-                            % self.friendship_id)
+        if self.is_ready():
+            return
+        self._payload = await self.puppet.friendship_payload(
+            friendship_id=self.friendship_id)
 
     def contact(self) -> Contact:
         """
         get the contact of the friendship
         """
         if self.puppet is None:
             raise Exception('puppet not found ...')
+        if not self.payload:
+            raise Exception('payload not ready ...')
```

Comment:
My mistake, I delete the code but added `await self.ready()` before, but delete that line later.

Because the `contact = self.wechaty.Contact.load(self.payload.contact_id)` line will somehow reports error on type check.

But still I think this code is not elegant enough, we can make a decorator to do these check on puppet and payload existance later.

Reply:
We will gradually make the code more and more elegant, so this module will be discussed later.